site/static/compare/script.js

```diff
@@ -709,4 +712,65 @@ function submitSettings() {
     window.location.search = params.toString();
 }
 
+function exportToMarkdown(testCases) {
```

Comment:
I know it might be more of a pain to add, but it feels like this should be done in Rust and exposed over an endpoint.

Reply:
Hmm, I'm not sure what benefits would that have :thinking: Both the summary table and the changes table are modeled after the compare page, which is presented in JavaScript.

What would be the motivation for this?

Reply:
You have a point - we're already splitting the logic for what summary metrics we show across JS and Rust (i.e., mean, range, count). It would be nice if all of this lived in one place, but this PR doesn't necessarily introduce that issue. And I suppose we can't really fix that without doing server-side rendering.

Reply:
It might be useful to make the summary table on the compare page and in PR comments should be as similar as possible, although I don't think that it's strictly necessary, the comment should contain (a possibly full) subset of the information on the compare page, since that should be more detailed.

Regarding the implementation, since the tables mostly differ in their presentation, I don't think that even backend rendering would solve this. Even if we generated some summary struct in Rust that would be used to render PR comments, it would then need to be rendered in various ways (HTML/CSS, Markdown). The closest thing to unify everything would probably be to generate Markdown on the server and then just render it as HTML on the compage page, but then we would lose a lot of the customization that we have there (colors, custom presentation etc.).

It's a good point, since this adds another place that needs to be changed if we modify the tables, but I consider this button to be mostly a helper tool for PR authors/experts, that will probably be hand-modified anyway.

For now I think that it's ok to leave it like this, to avoid complicating the code a lot because of a rather simple feature.